openedx/features/course_experience/tests/views/test_masquerade.py

```diff
@@ -0,0 +1,128 @@
+"""
+Tests for masquerading functionality on course_experience
+"""
+
+
+import json
+
+import six
+from django.urls import reverse
+
+from lms.djangoapps.commerce.models import CommerceConfiguration
+from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
+from openedx.features.course_experience import DISPLAY_COURSE_SOCK_FLAG, SHOW_UPGRADE_MSG_ON_COURSE_HOME
+from student.roles import CourseStaffRole
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from xmodule.partitions.partitions import ENROLLMENT_TRACK_PARTITION_ID
+from xmodule.partitions.partitions_service import PartitionService
+
+from .helpers import add_course_mode
+from .test_course_home import TEST_UPDATE_MESSAGE, course_home_url
+from .test_course_sock import TEST_VERIFICATION_SOCK_LOCATOR
+
+TEST_PASSWORD = 'test'
+UPGRADE_MESSAGE_CONTAINER = 'section-upgrade'
+
+
+class MasqueradeTestBase(SharedModuleStoreTestCase):
+    """
+    Base test class for masquerading functionality on course_experience
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(MasqueradeTestBase, cls).setUpClass()
+
+        # Create four courses
+        cls.verified_course = CourseFactory.create()
+        cls.masters_course = CourseFactory.create()
+        # Assign each verifiable course an upgrade deadline
+        add_course_mode(cls.verified_course, upgrade_deadline_expired=False)
```

Comment:
This code does more than create an upgrade deadline here. Do you think a comment like, "Create a verifiable course mode with an upgrade deadline in each course" would better describe what is happening in the following lines?